### PR TITLE
Usb late

### PR DESCRIPTION
**Pull request by CarlFK**
Mirroring pxe changes.

### CODE REVIEW

File: scripts/d-i/late_command.sh [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+set -eufx
+
+# This script setups ansible and runs it
+# It should be ran at the end of the basic installation of a machine
+
+apt install -y ansible git eatmydata
+
+# We clone our ansible repository and copy the ansible config files
+
+git clone https://github.com/CarlFK/video-stack-deploy.git /root/playbook-repo
```
Comment (stefanor) [inline scripts/d-i/late_command.sh]: These are appropriate for you, but not us. You were planning to template this, right?

Comment (CarlFK) [reply scripts/d-i/late_command.sh]: "yes" - really planning on using the pxe template.

File: scripts/mk_usb_installer.sh [hunk 2]
```diff
@@ -49,8 +50,9 @@ preseed="url=$(hostname):8000"
 # from the syslinux on the usb stick:
 # example: to install to an SSD that doesn't come up as /dev/sda:
 # (don't forget to escape the slasses because bash)
-# appends='partman-auto\/disk=\/dev\/nvme0n1 grub-installer\/bootdev=\/dev\/nvme0n1 hostname=gator domain=lan'
-appends='partman-auto\/disk=\/dev\/sda grub-installer\/bootdev=\/dev\/sda'
+# appends='partman-auto\/disk=\/dev\/sda grub-installer\/bootdev=\/dev\/sda'
+
+appends='partman-auto\/disk=\/dev\/sda grub-installer\/bootdev=\/dev\/sda hostname=testme domain=lan lc\/playbook_repo=https:\/\/github.com\/CarlFK\/video-stack-deploy.git lc\/playbook_branch=master lc\/inventory_repo=https:\/\/github.com\/xfxf\/av-foss-stack.git lc\/inventory_branch=master hw-detect\/load_firmware=false'
```
Comment (stefanor) [inline scripts/mk_usb_installer.sh]: This line needs to be broken up. And probably sourced from command line parsing? Or variables or a config file.

File: scripts/mk_usb_installer.sh [hunk 3]
```diff
@@ -32,7 +32,8 @@ dev=$1
 
 # easy: leave this as is.
 # it will use the server run at the end of this script.
-preseed="url=$(hostname):8000"
+# preseed="url=$(hostname):8000"
+preseed="url=10.100.7.247:8000"
```
Comment (stefanor) [inline scripts/mk_usb_installer.sh]: That dosen't seem like a good default.

File: scripts/d-i/xenial/preseed_video.cfg [hunk 4]
```diff
@@ -29,7 +29,7 @@ d-i preseed/early_command string cd /tmp && wget http://$url/ec/early_command.sh
 # d-i preseed/run string /ec/early_command.sh
 
 ### cfk - get late script from shaz, run it.
-d-i preseed/late_command string cd /target/tmp && unset http_proxy; wget http://$url/late_command.sh && chmod u+x late_command.sh && ANSIBLE_UNDER_DI=1 in-target /tmp/late_command.sh $url $(debconf-get mirror/suite) $(debconf-get passwd/username)
+d-i preseed/late_command string cd /target/tmp && unset http_proxy; wget http://$url/d-i/late_command.sh && chmod u+x late_command.sh && ANSIBLE_UNDER_DI=1 in-target /tmp/late_command.sh $url $(debconf-get mirror/suite) $(debconf-get passwd/username) $(debconf-get lc/playbook_repo) $(debconf-get lc/playbook_branch) $(debconf-get lc/inventory_repo) $(debconf-get lc/inventory_branch)
```
Comment (stefanor) [inline scripts/d-i/xenial/preseed_video.cfg:32]: hrm, what even is preseed-video.cfg?

File: scripts/d-i/stretch/preseed.cfg [hunk 5]
```diff
@@ -101,7 +101,7 @@ d-i mirror/country string manual
 d-i mirror/http/hostname string deb.debian.org
 d-i mirror/http/directory string /debian
 d-i mirror/http/proxy string
```
Comment (stefanor) [inline scripts/d-i/stretch/preseed.cfg:103]: Do we need this? Is it going to prompt, without it?

File: scripts/d-i/stretch/preseed.cfg [hunk 6]
```diff
@@ -101,7 +101,7 @@ d-i mirror/country string manual
...
-d-i mirror/http/proxy string http://dc10b:8000/
+# d-i mirror/http/proxy string http://dc10b:8000/
```
Comment (stefanor) [inline scripts/d-i/stretch/preseed.cfg:104]: Again, a bit of a weird default (even if it is a documenting comment)

File: scripts/d-i/late_command.sh [hunk 7]
```diff
@@ -0,0 +1,88 @@
...
+
+cd /home/$user
+
+# unset http_proxy
+# wget --recursive --level=0 --no-host-directories --no-parent \
+    # --reject "*.iso","*.gz",".*","*.swp","*.efi" \
+    # --reject-regex "carl/.*" \
+    # --default-page=.hideme \
+    # http://$server:8000/
+# #    --reject-regex ".*\.git/.*" \
```
Comment (stefanor) [inline scripts/d-i/late_command.sh]: :fire: ?

File: scripts/d-i/late_command.sh [hunk 8]
```diff
@@ -0,0 +1,88 @@
...
+
+git clone $playbook_repo /root/playbook-repo
+(cd /root/playbook-repo; git checkout $playbook_branch)
+# INVENTORY=/root/playbook-repo/inventory/hosts
+PLAYBOOKS=/root/playbook-repo/site.yml
+
+git clone $inventory_repo /root/inventory-repo
+(cd /root/inventory-repo; git checkout $inventory_branch)
+
+INVENTORY=/root/inventory-repo/inventory/hosts
```
Comment (stefanor) [inline scripts/d-i/late_command.sh]: What if there isn't an inventory-repo?

File: scripts/d-i/late_command.sh [hunk 9]
```diff
@@ -0,0 +1,88 @@
...
+# case $suite in
+#    stretch|stable)
+#        apt install dirmngr
+#        apt-key adv --recv-keys 93C4A3FD7BB9C367
+#        apt-add-repository --yes --update "deb http://ppa.launchpad.net/ansible/ansible/ubuntu xenial main"
+#    ;;
+#    xenial)
+#        apt-add-repository --yes --update ppa:ansible/ansible
+#    ;;
+#esac
```
Comment (stefanor) [inline scripts/d-i/late_command.sh]: Should there be an argument for an if block around this?

File: scripts/mk_usb_installer.sh (hunk 3 above)
Comment (CarlFK) [reply scripts/mk_usb_installer.sh]: oh yeah..  working around wonky network.  $(hostname) should work well some and maybe most  of the time ;)

ill fix

File: scripts/d-i/xenial/preseed_video.cfg (hunk 4 above)
Comment (CarlFK) [reply scripts/d-i/xenial/preseed_video.cfg:32]: my attempt at splitting up the pressed file into chunks that need swapping now and then. 

preseed.cfg just links to 3 or 4 more files.  If I remove the -disk one, then the installer will prompt me for disk parameters. 

Not sure how useful it really is.

File: scripts/d-i/stretch/preseed.cfg (hunk 6 above)
Comment (CarlFK) [reply scripts/d-i/stretch/preseed.cfg:104]: I need to move that to all the other parameters.

File: scripts/d-i/late_command.sh (hunk 7 above)
Comment (CarlFK) [reply scripts/d-i/late_command.sh]: What I was using to get a copy of the git repo from my laptop without committing.  Trying to test before commits.

File: scripts/d-i/late_command.sh (hunk 8 above)
Comment (CarlFK) [reply scripts/d-i/late_command.sh]: Make one :p 
I'll add some ifs.

File: scripts/d-i/stretch/preseed.cfg (hunk 5 above)
Comment (CarlFK) [reply scripts/d-i/stretch/preseed.cfg:103]: I think it will prompt without it.

File: scripts/d-i/late_command.sh (hunk 9 above)
Comment (CarlFK) [reply scripts/d-i/late_command.sh]: meh - it is commented out from when it seemed we needed a newer ansible.  somehow that requirement went away.  I left the code in as a hint on how to get newer ansible if the need comes up again.

File: scripts/d-i/late_command.sh [hunk 10]
```diff
@@ -0,0 +1,70 @@
...
+# It should be ran at the end of the basic installation of a machine
+
+# Here is where the parameters come from:
+
+# linux /debian/stretch/amd64/linux gfxpayload=800x600x16,800x600 --- auto=true url=dc10b DEBCONF_DEBUG=5 tasks="" hw-detect/load_firmware=false hostname= domain= interface=${net_default_mac} ${appends} lc/playbook_repo=carlfk lc/playbook_branch=master lc/inventory_repo=xfxf lc/inventory_branch=master
+
+# d-i preseed/late_command string cd /target/tmp && wget http://$url/d-i/late_command.sh && chmod u+x late_command.sh && ANSIBLE_UNDER_DI=1 in-target /tmp/late_command.sh $url $(debconf-get mirror/suite) $(debconf-get passwd/username) $(debconf-get lc/playbook_repo) $(debconf-get lc/playbook_branch) $(debconf-get lc/inventory_repo) $(debconf-get lc/inventory_branch)
+
+server=$1
+suite=$2
```
Comment (stefanor) [inline scripts/d-i/late_command.sh:15]: Those were only used by the debugging bits, that we removed.

**Review by stefanor — APPROVED** [review]

**Pull request merged**